src/ploigos_step_runner/utils/xml.py

```diff
@@ -87,3 +91,93 @@ def get_xml_element_by_path(xml_file_path, xpath, default_namespace=None, xml_na
         return xml_file.find(xpath)
 
     return xml_file.find(xpath, namespaces)
+
+def aggregate_xml_element_attribute_values(xml_file_path, xml_element, attribs): # pylint: disable=too-many-branches
+    """
+    Function to parse a XML file looking for a specific element and obtaining its attributes.
+
+    If element is not found in file the file is ignored.
+
+    If attribute is not found in element the file is ignored.
+
+    Parameters
+    ----------
+    xml_file_path: str
+        Path to directory containing XML files
+    xml_element: str
+        XML element being searched for
+    attribs: list
+        List of attributes being searched for in the XML element
+
+    Returns
+    --------
+    report_results: dict
+        A dictionary of attribute values.
+
+    Raises
+    ------
+    ValueError
+        If XML element is not found in file(s)
+
+    """
+
+    report_results = {}
+
+    #Search file path for XML files
+
+    if os.path.isdir(xml_file_path):
+        xml_files = glob.glob(xml_file_path + '/*.xml', recursive=False)
+    elif os.path.isfile(xml_file_path):
+        xml_files = [xml_file_path]
+
+    with open(xml_files[0], "r") as xml_file:
+        print("line: " + xml_file.read())
+
+    for attrib in attribs:
+        report_results[attrib] = []
+
+    #Iterate over XML files
+    for file in xml_files:
+        try:
+            pom_element = get_xml_element(file, xml_element)
+            #Add attributes to dictionary
+            for attrib in attribs:
+                attribute_value = pom_element.attrib[attrib]
+                # aggregate attribute accross multiple files
+                report_results[attrib].append(attribute_value)
+        except ValueError:
+            #Return empty dict if an attribute is missing
+            return {}
+
+    #Loop to check if all values aggregated are integers, floats, strings, or a mix
+    #of all three.
+    for attrib in attribs:
+        only_numerics = True
+        only_floats = True
+        #Iteration to determine what pattern all values follow.
+        for value in report_results[attrib]:
+            if only_numerics and value.isnumeric():
+                pass
```

Comment:
I'm thinking this can be changed to a `continue` - if it is numeric, there's no need to check if it converts to float, so we can just skip ahead to the next loop

Reply:
So isnumeric is kind of weird in that floats will return False for that function.

Reply:
A value could be both numeric and a float, but not necessarily a float and numeric.

Reply:
Nevermind I see what you mean. If it's numeric then yeah it has to be a valid float.

Reply:
Right - if isnumeric returns false it keeps running and tries the float(), but if it's true, it should always pass the float() test... if I understand isnumeric correctly

Reply:
And numerics are taking precedence over floats later on anyway.

Reply:
The way I understand it is isnumeric will be true for positive integers

Reply:
Nvm, didn't see your response earlier

Reply:
You have it correct my mind was working backwards on me.